packages/workers-playground/src/QuickEditor/HTTPTab/ResponseView.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { Div } from '@cloudflare/elements';
+import CodeBlock from '@cloudflare/component-code-block';
+type Props = {
+  response: Response;
+  loading: boolean;
+};
+
+const StatusIndicator: React.FC<{ statusCode: number }> = ({ statusCode }) => {
+  const backgroundColor =
+    statusCode < 300 ? 'green.5' : statusCode < 400 ? 'gold.6' : 'red.4';
+  return (
+    <Div
+      display="inline-block"
+      width=".8em"
+      height=".8em"
+      mr={1}
+      borderRadius="50%"
+      backgroundColor={backgroundColor}
+    />
+  );
+};
+
+function maybeGetLanguage(
+  headers: string[][]
+): Parameters<typeof CodeBlock>[0]['language'] {
+  const contentType = headers.find((h) => h[0] === 'content-type')?.[1] ?? '';
+  switch (contentType) {
+    case 'text/html':
+      return 'html';
+    case 'application/javascript':
+      return 'javascript';
+    case 'application/json':
+      return 'json';
+    case 'text/css':
+      return 'css';
+    default:
+      return 'txt';
+  }
+}
+
+const ResponseView: React.FC<Props> = ({ response, loading }) => {
+  const { status, statusCode } = useMemo(() => {
+    const status = response.headers.get('cf-ew-status') || '';
+    const statusCode = window.parseInt(status, 10);
+    return { status, statusCode };
+  }, [response]);
+
+  const headers = useMemo(
+    () =>
+      [...response.headers]
+        .filter(([header]) => header.startsWith('cf-ew-raw-'))
+        .map(([header, value]) => [header.replace('cf-ew-raw-', ''), value]),
+    [response]
+  );
+
+  const [body, setBody] = useState('');
+
+  useEffect(() => {
+    void response.text().then(setBody);
```

Comment:
It's pretty unlikely that you'd get a new response before the body is consumed, but it shouldn't matter if you do. `response` is stored in React state, and so will be replaced if a new one comes in. In theory there's a race condition here where the body is set out of order, but since these come from user interaction (and the button is disabled while a request is loading) I don't think that will be a concern.